test/test_retrieve_filepath.py

```diff
@@ -0,0 +1,69 @@
+"""
+Test the utility function found in lib/DataFileUtil/utils/retrieve_filename.py
+
+We use example URLs pulled from DataFileUtil_server_test.py
+
+In the future, we could use the `responses` URL mocking package instead.
+"""
+import unittest
+from uuid import uuid4
+
+from DataFileUtil.utils.retrieve_filename import retrieve_filename
+
+
+class RetrieveFilepathTest(unittest.TestCase):
+
+    def test_invalid_url(self):
+        url = "xyz"
+        with self.assertRaises(ValueError) as ctx:
+            retrieve_filename(url)
+        ex = str(ctx.exception)
+        msg = f"Cannot connect to URL: {url}"
+        self.assertTrue(msg in ex)
+
+    def test_no_path_no_header(self):
+        """Test the case where there is no filepath and no filename header."""
+        url = "https://www.google.com"
+        fn = retrieve_filename(url)
+        # Assert that filename is a non-empty string
+        self.assertTrue(isinstance(fn, str))
+        self.assertTrue(len(fn) > 0)
+
+    def test_use_url_filepath(self):
+        """
+        Test the case where we cannot fetch from the URL, so
```

Comment:
If you can't fetch from the url shouldn't that cause an exception?

Reply:
Yes that would cause an exception. This URL is fetchable, but there is no content header. Now that I'm looking at it, this test is redundant with one below.

Reply:
Ok, that's what I was thinking, but I wasn't sure, and the language confused me

Reply:
Removed the redundant test